packages/camera/camera_avfoundation/example/ios/RunnerTests/AvailableCamerasTest.m

```diff
@@ -55,29 +54,26 @@ - (void)testAvailableCamerasShouldReturnAllCamerasOnMultiCameraIPhone {
   }
   OCMStub([discoverySessionMock devices]).andReturn([NSArray arrayWithArray:cameras]);
 
-  // Set up method call
-  FlutterMethodCall *call = [FlutterMethodCall methodCallWithMethodName:@"availableCameras"
-                                                              arguments:nil];
-
-  __block id resultValue;
-  [camera handleMethodCallAsync:call
-                         result:^(id _Nullable result) {
-                           resultValue = result;
-                           [expectation fulfill];
-                         }];
+  __block NSArray<FCPPlatformCameraDescription *> *resultValue;
+  [camera
+      availableCamerasWithCompletion:^(NSArray<FCPPlatformCameraDescription *> *_Nullable result,
+                                       FlutterError *_Nullable error) {
+        XCTAssertNil(error);
+        resultValue = result;
+        [expectation fulfill];
+      }];
+  [self waitForExpectationsWithTimeout:30 handler:nil];
```

Comment:
I'm not sure why we weren't actually using the expectation before, but now it's necessary because the queue dispatch isn't being bypassed in the test like it was before, so the block above is no longer run synchronously in the test.